src/plugins/intel_gpu/src/kernel_selector/cl_kernels/broadcast_gpu_opt.cl

```diff
@@ -0,0 +1,198 @@
+    const uint out_f  = (out_bfwz / OUTPUT_SIZE_Z) % OUTPUT_FEATURE_NUM;
+    const uint out_b  = (out_bfwz / OUTPUT_SIZE_Z / OUTPUT_FEATURE_NUM);
+#else
+    const uint out_bfwz = (uint) get_global_id(2);
+    const uint out_z  = 0;
+    const uint out_w  = 0;
+    const uint out_f  = (out_bfwz % OUTPUT_FEATURE_NUM);
+    const uint out_b  = (out_bfwz / OUTPUT_FEATURE_NUM);
+#endif
+
+    const uint in_x = out_x % in_sx;
+    const uint in_y = out_y % in_sy;
+    const uint in_f = out_f % in_sf;
+    const uint in_b = out_b % in_sb;
+
+#if INPUT0_DIMS == 6
+    const uint in_w = out_w % in_sw;
+    const uint in_z = out_z % in_sz;
+    const uint in_pos =  INPUT0_OFFSET + in_x + in_sx * (in_y + in_sy * (in_z + in_sz * (in_w + in_sw * (in_f + in_sf * in_b))));
+#elif INPUT0_DIMS == 5
+    const uint in_z = out_z % in_sz;
+    const uint in_pos =  INPUT0_OFFSET + in_x + in_sx * (in_y + in_sy * (in_z + in_sz * (in_f + in_sf * in_b)));
+#else
+    const uint in_pos =  INPUT0_OFFSET + in_x + in_sx * (in_y + in_sy * (in_f + in_sf * in_b));
+#endif
+
+    uint idx[INPUT0_DIMS] = {0};
+    uint rmd = in_pos;
+    unroll_for (int i = 0; i < INPUT0_DIMS; ++i)
+    {
+        idx[i] = rmd / blockND[i + 1];
+        rmd %= blockND[i + 1];
+    }
+    const uint idx_b = idx[0];
+    const uint idx_f = idx[1];
+    #if INPUT0_DIMS == 4
+        const uint idx_y = idx[2];
+        const uint idx_x = idx[3];
+    #elif INPUT0_DIMS == 5
+        const uint idx_z = idx[2];
+        const uint idx_y = idx[3];
+        const uint idx_x = idx[4];
+    #elif INPUT0_DIMS == 6
+        const uint idx_w = idx[2];
+        const uint idx_z = idx[3];
+        const uint idx_y = idx[4];
+        const uint idx_x = idx[5];
+    #endif
+
+    const uint idx_pos = GET_UPDATES_INDEX(INPUT0, IDX_ORDER);
+#if OUTPUT_DIMS == 6
+    const uint out_pos_base = OUTPUT_GET_INDEX(out_b, out_f, out_w, out_z, out_y, out_x);
+#elif OUTPUT_DIMS == 5
+    const uint out_pos_base = OUTPUT_GET_INDEX(out_b, out_f, out_z, out_y, out_x);
+#else
+    const uint out_pos_base = OUTPUT_GET_INDEX(out_b, out_f, out_y, out_x);
+#endif
+
+    if (use_opt_code) {
+        uint out_pos = out_pos_base;
+        INPUT0_VTYPE input_vec = VLOAD(0, &input[idx_pos]);
+        unroll_for(uint i = 0; i < Y_BLOCKS; i++) {
+            VSTORE(input_vec, 0, &output[out_pos]);
+            out_pos += OUTPUT_SIZE_X;
+        }
+        if (gdim1 < y_leftovers) {
+            VSTORE(input_vec, 0, &output[out_pos]);
+        }
+
+        if (gdim0 < x_leftovers) {
+            INPUT0_TYPE input_val = input[idx_pos + VEC_SIZE];
+
+            out_pos = out_pos_base;
+            unroll_for(uint i = 0; i < Y_BLOCKS; i++) {
+                output[out_pos + VEC_SIZE] = input_val;
+                out_pos += OUTPUT_SIZE_X;
+            }
+
+            if (gdim1 < y_leftovers)
+                output[out_pos + VEC_SIZE] = input_val;
+        }
+    } else {
+        output[out_pos_base] = input[idx_pos];
+    }
+}
+
```

Comment:
Random spot:
Seems that currently any 5D broadcast can select this kernel and do the opt code if it is y axis broadcast and do the ref code otherwise. If so, why don't we add this additional opt case to the ref kernel instead of creating a new kernel?

Reply:
Done. please review it again. thanks.